employees/views.py

```diff
@@ -67,108 +63,60 @@ def perform_create(self, serializer: ReportSerializer) -> None:
+                return self.form_valid_for_join_project()
+            else:
+                return self.form_invalid_for_join_project(form)
         else:
-            reports_serializer = ReportSerializer(context={"request": self.request})
-            reports_serializer.fields["date"].initial = str(datetime.datetime.now().date())
-        reports_serializer.fields["project"].queryset = Project.objects.filter(
-            members__id=self.request.user.id
-        ).order_by("name")
-        reports_serializer.fields["task_activities"].queryset = TaskActivityType.objects.order_by("name")
-        return reports_serializer
-
-    def _create_project_join_form(self) -> ProjectJoinForm:
-        project_form_queryset = Project.objects.exclude(members__id=self.request.user.id).order_by("name")
-        self.hide_join = not project_form_queryset.exists()
-        return ProjectJoinForm(queryset=project_form_queryset)
-
-    def initial(self, request: HttpRequest, *args: Any, **kwargs: Any) -> None:
-        logger.info(f"Initial method for user with id: {self.request.user.pk} in ReportList view")
-        super().initial(request, *args, **kwargs)
-        self.reports = self.get_queryset()
-        self.daily_hours_sum = self.reports.order_by().get_work_hours_sum_for_all_dates()
-        self.monthly_hours_sum = self.reports.order_by().get_work_hours_sum_for_all_authors()
-
-    def get(self, _request: HttpRequest) -> Response:
-        logger.info(f"User with id: {self.request.user.pk} get to the ReportList view")
-        return Response(
-            {
-                "serializer": self._create_serializer(),
-                "object_list": self.reports,
-                "daily_hours_sum": self.daily_hours_sum,
-                "monthly_hours_sum": self.monthly_hours_sum,
-                "UI_text": ReportListStrings,
-                "project_form": self._create_project_join_form(),
-                "hide_join": self.hide_join,
-            }
+            return super().post(request, *args, **kwargs)
+
+    def get_form_kwargs(self) -> dict:
+        kwargs = super().get_form_kwargs()
+        kwargs["user"] = self.request.user
+        return kwargs
+
+    def get_context_data(self, **kwargs: Any) -> dict:
+        context_data = super().get_context_data(**kwargs)
+        context_data["object_list"] = (
+            super().get_queryset().filter(author=self.request.user).order_by("-date", "project__name", "-creation_date")
+        )
+        context_data["daily_hours_sum"] = context_data["object_list"].order_by().get_work_hours_sum_for_all_dates()
+        context_data["monthly_hours_sum"] = context_data["object_list"].order_by().get_work_hours_sum_for_all_authors()
+        context_data["project_form"] = ProjectJoinForm(
+            queryset=Project.objects.exclude(members__id=self.request.user.id).order_by("name")
         )
+        context_data["hide_join"] = not Project.objects.exclude(members__id=self.request.user.id).exists()
```

Comment:
This looks a little bit over complicated, it should be enough to do
```
Project.objects.filter(members=self.request.user).exists()
```

Reply:
I am not sure, because we want to hide join button where there is no project to join by current user. This will return true even when user will be in one project and it will hide join button and show message `No project to join`.

Reply:
Its OK then, thanks for explanation.